rs/nns/governance/src/governance.rs

```diff
@@ -6762,13 +6766,51 @@ impl Governance {
                     );
                 }
             }
-            self.schedule_pending_rewards_distribution(day_after_genesis, reward_distribution);
+            Some(reward_distribution)
+        };
+
+        let reward_event = RewardEvent {
+            day_after_genesis,
+            actual_timestamp_seconds: now,
+            settled_proposals: considered_proposals,
+            distributed_e8s_equivalent: actually_distributed_e8s_equivalent,
+            total_available_e8s_equivalent: total_available_e8s_equivalent_float as u64,
+            rounds_since_last_distribution: Some(rounds_since_last_distribution),
+            latest_round_available_e8s_equivalent: Some(
+                latest_round_available_e8s_equivalent_float as u64,
+            ),
+        };
+
+        Some((reward_event, reward_distribution))
+    }
+
+    /// Create a reward event.
```

Comment:
This seems like NOT the main point of this method. Yes, that is one of the effects. That is just a recording of a SUMMARY of what took place. The ACTUALLY INTERESTING thing that this does is increase the maturity of neurons according to how much they recently voted. This is the lede. Always lead with the lede!